homeassistant/components/venstar/sensor.py

```diff
@@ -116,6 +158,34 @@ def native_value(self) -> float:
         return round(float(self._client.get_sensor(self.sensor_name, "temp")), 1)
 
 
+class VenstarBatterySensor(VenstarSensor):
+    """Represent a Venstar battery sensor."""
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return f"{self._client.name} {self.sensor_name} Battery"
+
+    @property
+    def native_value(self) -> int:
+        """Return state of the sensor."""
+        return self._client.get_sensor(self.sensor_name, "battery")
+
+
+class VenstarRuntimeSensor(VenstarSensor):
+    """Represent a Venstar runtime sensor."""
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return f"{self._client.name} {RUNTIME_ATTRIBUTES[self.sensor_name]} Runtime"
+
+    @property
+    def native_value(self) -> int:
+        """Return state of the sensor."""
+        return self.coordinator.runtimes[-1][self.sensor_name]
```

Comment:
If we define lambda functions in the descriptions we don't need these different entity classes. We can use the entity descriptions to differentiate the entities. For the native value, the lambda function can accept the coordinator as a parameter and return whatever data needed. For the entity name, the lambda can accept the client and the sensor name.

Example:
https://github.com/home-assistant/core/blob/caa04c186699fef953e8fb5840f5cac8e7bb016e/homeassistant/components/airly/sensor.py#L121